apps/web/src/components/Table/ResponsiveTable/RowSelection.tsx

```diff
@@ -146,6 +146,55 @@ export const actionButtonStyles: Pick<
   fontSize: "caption",
 };
 
+interface DownloadAllButtonProps {
+  download: DownloadDef["all"];
+}
+
+const DownloadAllButton = ({ download }: DownloadAllButtonProps) => {
+  const intl = useIntl();
+  if (!download) return null;
+
+  const label =
+    "label" in download && download?.label
+      ? download.label
+      : intl.formatMessage({
+          defaultMessage: "Download full dataset",
+          id: "B6XXtf",
+          description:
+            "Text label for button to download a csv file of all items in a table.",
+        });
+
+  if ("csv" in download) {
+    return (
+      <DownloadCsv {...download.csv} {...actionButtonStyles}>
+        {label}
+      </DownloadCsv>
+    );
+  }
+
+  if ("component" in download && download.component) {
+    return download.component;
+  }
+
+  if ("onClick" in download) {
+    return (
+      <Button
+        {...actionButtonStyles}
+        onClick={download.onClick}
+        disabled={download.downloading}
+        data-h2-font-weight="base(400)"
+        {...(download.downloading && {
+          icon: SpinnerIcon,
+        })}
+      >
+        {label}
+      </Button>
+    );
+  }
+
+  return null;
+};
```

Comment:
This is messy but hopefully we can clean it up if we move the skills download to the back end as well.